templates/components/multiagent/python/app/api/routers/chat.py

```diff
@@ -0,0 +1,46 @@
+import logging
+
+from app.api.routers.events import EventCallbackHandler
+from app.api.routers.models import (
+    ChatData,
+)
+from app.api.routers.vercel_response import VercelStreamResponse
+from app.engine import get_chat_engine
+from fastapi import APIRouter, BackgroundTasks, HTTPException, Request, status
+
+chat_router = r = APIRouter()
+
+logger = logging.getLogger("uvicorn")
+
+
+@r.post("")
+async def chat(
+    request: Request,
+    data: ChatData,
+    background_tasks: BackgroundTasks,
+):
+    try:
+        last_message_content = data.get_last_message_content()
+        messages = data.get_history_messages(include_agent_messages=True)
+
+        event_handler = EventCallbackHandler()
+        # The chat API supports passing private document filters and chat params
+        # but agent workflow does not support them yet
+        # ignore chat params and use all documents for now
+        # TODO: generate filters based on doc_ids
+        # TODO: use chat params
+        engine = get_chat_engine(chat_history=messages)
+
+        event_handler = engine.run(input=last_message_content, streaming=True)
+        return VercelStreamResponse(
```

Comment:
@leehuwuj cool, we're using now `VercelStreamResponse` for both workflow and chat engine?

Reply:
> we're using now VercelStreamResponse for both workflow and chat engine?

Yes, i refactored the code to use a same class name (to avoid importing issue) but please note that the parameters are still different 😃